### PR TITLE
Add a constructor to `Stripe` to impose invariants early

### PR DESCRIPTION
**Pull request by JosiahWI**
The recent commits are not signed because gpg broke on WSL after an update.

This change also revealed an issue with the `Stripe` unit tests. The write head was pointing to a hardcoded value that happened to be valid instead of the start of the document blocks. This also fixes that issue - see c39be53.

### CODE REVIEW

**Comment by JosiahWI** [comment]
The AuTest post-early-return failed.

File: src/iocore/cache/StripeSM.cc [hunk 1]
```diff
@@ -151,36 +162,11 @@ StripeSM::clear_dir()
...
-  // Hash
-  char        *seed_str       = disk->hash_base_string ? disk->hash_base_string : s;
-  const size_t hash_seed_size = strlen(seed_str);
-  const size_t hash_text_size = hash_seed_size + 32;
-
-  hash_text = static_cast<char *>(ats_malloc(hash_text_size));
-  ink_strlcpy(hash_text, seed_str, hash_text_size);
-  snprintf(hash_text + hash_seed_size, (hash_text_size - hash_seed_size), " %" PRIu64 ":%" PRIu64 "",
-           static_cast<uint64_t>(dir_skip), static_cast<uint64_t>(blocks));
-  CryptoContext().hash_immediate(hash_id, hash_text, strlen(hash_text));
```
Comment (masaori335) [inline src/iocore/cache/StripeSM.cc:165]: Where this code goes?

Comment (JosiahWI) [reply src/iocore/cache/StripeSM.cc:165]: It moved to `Stripe::_init_hash_text`.

Comment (JosiahWI) [reply src/iocore/cache/StripeSM.cc:165]: It happened in 0412927.

Comment (masaori335) [reply src/iocore/cache/StripeSM.cc:165]: do you mean `Stripe::_init_hash_text` function? I can see `snprintf` makes the `hash_text`, but I can't find using it for `hash_id`. What I'm missing?
https://github.com/apache/trafficserver/pull/11811/files#diff-88d9e07d3781ce4d9c6307114be7856582eba25edc0d1c64ac9a73a59d00baecR102-R113

**Review by masaori335 — APPROVED** [review]
Looks good! Thanks for cleaning up.

**Pull request merged**